svm/src/transaction_processing_result.rs

```diff
@@ -0,0 +1,40 @@
+use {
+    crate::transaction_execution_result::ExecutedTransaction,
+    solana_sdk::transaction::Result as TransactionResult,
+};
+
+pub type TransactionProcessingResult = TransactionResult<ProcessedTransaction>;
+pub type ProcessedTransaction = ExecutedTransaction;
+
+pub trait TransactionProcessingResultExtensions {
+    fn was_processed(&self) -> bool;
+    fn processed_transaction(&self) -> Option<&ProcessedTransaction>;
+    fn processed_transaction_mut(&mut self) -> Option<&mut ProcessedTransaction>;
+    fn flattened_result(&self) -> TransactionResult<()>;
+}
+
+impl TransactionProcessingResultExtensions for TransactionProcessingResult {
+    fn was_processed(&self) -> bool {
+        self.is_ok()
+    }
```

Comment:
You are considering not processed if self is err, but you also replaced `NotExecuted` by `Err`. Are you accounting for the case where a processed transaction may have not been executed?

Reply:
Not yet in this change. That change to support processing transactions that may not have been executed will come next in https://github.com/anza-xyz/agave/pull/2425. See https://github.com/anza-xyz/agave/pull/2425/files#diff-5eb12bc3fc00b926c18e4204f316d650074ce4fe4bce7d664459c3cf340af5aeR21-R24